src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -836,6 +889,19 @@ public function offsetGet($option)
             }
         }
 
+        // Check whether the option is deprecated
+        if (isset($this->deprecated[$option])) {
+            $deprecationMessage = $this->deprecated[$option];
+
+            if ($deprecationMessage instanceof \Closure) {
+                $deprecationMessage = $deprecationMessage($value);
+            }
+
+            if (\is_string($deprecationMessage)) {
```

Comment:
I would make it more strict: a string triggers a deprecation, an empty string does nothing, everything else throws an exception as not valid.

Reply:
I made the changes according to this.